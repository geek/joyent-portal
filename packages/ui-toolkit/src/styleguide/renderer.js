import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import remcalc from 'remcalc';
import is from 'styled-is';

import { Grid, Row, Col } from '../grid';
import theme from '../theme';

const Main = styled(Row)`
  padding-top: ${remcalc(40)};
`;

const Sticky = styled.div`
  position: sticky;
  top: 0;
  z-index: 9;

  ${is('sidebar')`
    top: ${remcalc(100)};
    max-height: 85vh;
    overflow: auto;
  `};
`;

const fullTheme = {
  ...theme,
  spacing: {
    0.5: remcalc(4),
    0: remcalc(0),
    1: remcalc(6),
    2: remcalc(12),
    3: remcalc(18),
    4: remcalc(24),
    5: remcalc(30),
    6: remcalc(36),
    7: remcalc(42),
    8: remcalc(48),
    9: remcalc(54),
    10: remcalc(60)
  }
};

const StyleGuideRenderer = ({
  title,
  homepageUrl,
  children,
  toc,
  hasSidebar
}) => (
  <ThemeProvider theme={fullTheme}>
    <Grid>
      <Main>
        {hasSidebar && (
          <Col xs={3}>
            <Sticky sidebar>{toc}</Sticky>
          </Col>
        )}
        <Col xs={hasSidebar ? 9 : 12} lg={hasSidebar ? 8 : 12}>
          {children}
        </Col>
      </Main>
    </Grid>
  </ThemeProvider>
);

export default StyleGuideRenderer;
