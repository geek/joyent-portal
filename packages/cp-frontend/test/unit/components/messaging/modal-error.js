/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import ModalError from '@components/messaging/modal-error';

it('renders <ModalError /> without throwing', () => {
  const tree = renderer
    .create(
      <ModalError message='Modal error message' onCloseClick={() => {}}/>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
