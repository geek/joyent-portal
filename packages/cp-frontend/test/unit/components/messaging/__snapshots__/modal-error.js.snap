// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <ModalError /> without throwing 1`] = `
.c5 {
  font-family: sans-serif;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  -moz-appearance: button;
  appearance: button;
  min-width: 7.5rem;
  box-sizing: border-box;
  display: inline-block;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
  padding: 0.9375rem 1.125rem;
  position: relative;
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  font-weight: 400;
  font-size: 0.9375rem;
  text-align: center;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-decoration: none;
  white-space: nowrap;
  vertical-align: middle;
  touch-action: manipulation;
  cursor: pointer;
  color: ;
  background-image: none;
  background-color: ;
  border-radius: 0.25rem;
  border: solid 0.0625rem;
  color: ;
  background-color: ;
  border-color: ;
}

.c5::-moz-focus-inner,
.c5[type="button"]::-moz-focus-inner,
.c5[type="reset"]::-moz-focus-inner,
.c5[type="submit"]::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c5:-moz-focusring,
.c5[type="button"]:-moz-focusring,
.c5[type="reset"]:-moz-focusring,
.c5[type="submit"]:-moz-focusring {
  outline: 0.0625rem dotted ButtonText;
}

.c5:focus {
  outline: 0;
  text-decoration: none;
  background-color: ;
  border-color: ;
}

.c5:hover {
  background-color: ;
  border: solid 0.0625rem;
}

.c5:active,
.c5:active:hover,
.c5:active:focus {
  background-image: none;
  outline: 0;
  background-color: ;
  border-color: ;
}

.c5[disabled] {
  cursor: not-allowed;
  pointer-events: none;
}

.c5:focus {
  background-color: ;
  border-color: ;
}

.c5:hover {
  background-color: ;
  border-color: ;
}

.c5:active,
.c5:active:hover,
.c5:active:focus {
  background-color: ;
  border-color: ;
}

.c5 + button {
  margin-left: 1.25rem;
}

.c2 {
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  font-weight: 500;
  font-size: 1.5rem;
}

.c4 {
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  font-weight: 400;
  line-height: 1.5rem;
  font-size: 0.9375rem;
}

.c1 {
  line-height: 1.25;
  color: ;
  margin: 0 0 0.75rem 0;
}

.c3 {
  color: ;
  margin: 0.75rem 0 1.875rem 0;
}

.c0 {
  color: ;
}

<div>
  <h2
    className="c0 c1 c2"
  />
  <p
    className="c3 c4"
  >
    Modal error message
  </p>
  <button
    className="c5"
    onClick={[Function]}
  >
    Close 
  </button>
</div>
`;
